sentence.go

```diff
@@ -60,21 +60,26 @@ func parseSentence(raw string) (BaseSentence, error) {
 	if startIndex != 0 {
 		return BaseSentence{}, fmt.Errorf("nmea: sentence does not start with a '$' or '!'")
 	}
-	sumSepIndex := strings.Index(raw, ChecksumSep)
-	if sumSepIndex == -1 {
-		return BaseSentence{}, fmt.Errorf("nmea: sentence does not contain checksum separator")
+
+	var checksumRaw string
+	splice := strings.Split(raw[startIndex+1:], ChecksumSep)
+	fieldsRaw := splice[0]
+	fields := strings.Split(fieldsRaw, FieldSep)
+
+	if len(splice) == 2 {
+		checksumRaw = strings.ToUpper(splice[1])
 	}
-	var (
-		fieldsRaw   = raw[startIndex+1 : sumSepIndex]
-		fields      = strings.Split(fieldsRaw, FieldSep)
-		checksumRaw = strings.ToUpper(raw[sumSepIndex+1:])
-		checksum    = xorChecksum(fieldsRaw)
-	)
-	// Validate the checksum
-	if checksum != checksumRaw {
-		return BaseSentence{}, fmt.Errorf(
-			"nmea: sentence checksum mismatch [%s != %s]", checksum, checksumRaw)
+
+	if checksumRaw != "" {
```

Comment:
This check seem redundant, the contents of this block can go into the `if len(splice) == 2` body.

Reply:
That makes sense, I was busy trying to figure something else out and rewrote it twice